pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -768,10 +784,72 @@ func (r *ReconcileBareMetalHost) actionManageReady(prov provisioner.Provisioner,
 	return r.manageHostPower(prov, info)
 }
 
-func (r *ReconcileBareMetalHost) saveStatus(host *metal3v1alpha1.BareMetalHost) error {
+func (r *ReconcileBareMetalHost) saveHostStatus(host *metal3v1alpha1.BareMetalHost) error {
 	t := metav1.Now()
 	host.Status.LastUpdated = &t
-	return r.client.Status().Update(context.TODO(), host)
+
+	if err := r.saveHostAnnotation(host); err != nil {
+		return err
+	}
+
+	//Refetch host again
+	obj := host.Status.DeepCopy()
+	err := r.client.Get(context.TODO(),
```

Comment:
Does this work? I thought the client went to the cache and the cache was updated right before Reconcile() is called.

Reply:
Yes, because the cache was updated in saveHostAnnotation() function once to save the annotation is called in line 791.